### PR TITLE
Userprofile page

### PR DESCRIPTION
**Pull request by Rjjayy**
So far, the User should be able to see their name, username, Bio, and an avatar.

### CODE REVIEW

File: package.json
Comment (akosasante) [inline package.json]: We shouldn't need this line since we are using styled-components library for styling with MaterialUI instead of emotion

File: package.json
Comment (akosasante) [inline package.json]: We shouldnt need this line since we're using styled-components

File: package.json
Comment (akosasante) [inline package.json]: So far, the other front end members are using just MaterialUI instead of the sister library JoyUI. We should stick to that for now so that all of the UI looks consistent unless we find a specific component or something thats only available in JoyUI

File: package.json
Comment (akosasante) [inline package.json]: Are you using Redux in this PR? I haven't had a chance to look at all of the changes, so maybe you are. Otherwise, I would suggest we hold off on bringing in Redux unless we really find a use case for it. React has built-in state management tooling like React Contexts and useReducer hooks that we can stick to unless we find we need something that handles  more complexity.  Or if it's just a tool you want to practice using more, that's fine as well. Just make sure to discuss it with your teammates as well.

File: package.json
Comment (akosasante) [inline package.json]: Is there a particular reason to use this older version of the library?

File: package.json
Comment (akosasante) [inline package.json]: Is there a reason you're removing this?

File: package.json
Comment (akosasante) [inline package.json]: Is there a reason to use this older version of the library?

**Review by akosasante — CHANGES_REQUESTED** [review]
I havent had a chance to look at the UserProfile yet but leaving some comments about the package.json changes

File: package.json
Comment (Rjjayy) [reply package.json]: Most of these changes were removed long before I pushed. My current code looks like the red that it stated I removed. Expect for the "react-redux" which I was thinking about using but never actually used it in the code. I will try pushing again.

**Comment by Rjjayy** [comment]
Undoing the changes made to package.json.

File: package-lock.json
Comment (akosasante) [inline package-lock.json:1]: This file was deleted; we don't want to delete it in main. Please run `npm install` locally and then commit and push the resulting `package-lock.json` file

File: package.json
Comment (akosasante) [inline package.json:1]: looks good 👍

File: public/index.html
Comment (akosasante) [inline public/index.html]: This already exists on line 6; we can delete this here or the earlier one instead of having both.

File: src/UserProfile
Comment (akosasante) [inline src/UserProfile:1]: This file (without `.js` extension) is empty; can you make sure it's not getting pushed to main. Delete it locally and if you're not seeing it locally, you may need to do `git rm -r --cached src/UserProfile`

File: src/UserProfile.js
Comment (akosasante) [inline src/UserProfile.js]: `useHistory` is not available in the version of react-router-dom that we're using: https://reactrouter.com/en/main/upgrading/v5#use-usenavigate-instead-of-usehistory

You'll want to use `useNavigate` instead.

File: src/UserProfile.js
Comment (akosasante) [inline src/UserProfile.js]: This is fine for now, but we'll probably want to mock using `mirage.js` in the future so that we at least have the code set up for making the API call to the backend. Masooma wrote some docs here: https://github.com/Code-the-Dream-School/dd-prac-team4-front/blob/main/docs/DOCUMENTATION.md#mirage-library-for-mocking-backend

File: src/UserProfile.js
Comment (akosasante) [inline src/UserProfile.js]: Will need to be updated to use `useNavigate`

File: src/UserProfile.js
Comment (akosasante) [inline src/UserProfile.js]: I don't know if @nikkigraybeal or @Shauna2022 might know more but this wasn't working for me locally and in the Material-UI docs it seems to indicate that `makeStyles` is legacy / maybe doesn't work the same now.

Comment (Shauna2022) [reply src/UserProfile.js]: it could be due to the import on your end? do you have the most up to date code on the front end?

Comment (Shauna2022) [reply src/UserProfile.js]: after a second look i see why its not working. i dont see the file that they are trying to import from

Comment (akosasante) [reply src/UserProfile.js]: @Rjjayy do you have `@mui/styles` installed locally? I was not able to get it working myself, it [says](https://github.com/mui/material-ui/issues/32142) that it's not compatible with React v18 which is what we're using. I'm get curious if there was something you did on your end to get this to work / were you able to render and view your component somehow?

Comment (akosasante) [reply src/UserProfile.js]: If it seems like it'll be difficult to get it to work; I think our recommendation would be to just keep it simple for now and use the unstyled components; or maybe just use plain CSS (or you can use [CSS modules](https://mui.com/material-ui/guides/interoperability/#css-modules)). We can revisit getting the styling working with MUI as a stretch goal but we should try to focus on getting some basic components up and the functionality on the frontend going as we get into the next sprint

**Comment by akosasante** [comment]
@Rjjayy  ; it's a bit difficult for me to test your changes since you `forked` the library instead of `cloning` it per the initial instructions in the repo [readme](https://github.com/Code-the-Dream-School/dd-prac-team4-front) and team meeting notes :cry: . For future tickets I would encourage you to make your PRs directly to this repo. If you don't want to remove your existing fork (which is totally fine!) you can do this by adding a second remote to your local repo:

```
git remote add team_repo https://github.com/Code-the-Dream-School/dd-prac-team4-front.git
```

Then whenever you push and are ready for it to be reviewed, you can do:

```
git push team_repo
```

**Comment by akosasante** [comment]
@Rjjayy Please take a look at my branch for a possible approach you can take to get things working with CSS modules and/or using the current Material-UI theming functionality: https://github.com/Code-the-Dream-School/dd-prac-team4-front/compare/main...akos-branch-raheim

File: package-lock.json
Comment (akosasante) [reply package-lock.json:1]: @Rjjayy can you double check and make sure you've done `git add package-lock.json` and then commited and pushed that? It's till showing as deleted here.

File: public/index.html
Comment (akosasante) [reply public/index.html]: bump @Rjjayy

File: src/App.js
Comment (akosasante) [inline src/App.js:32]: I think you noticed already; but if this is being added, we need to make sure that there's an import at the top of the file too

File: src/components/UserProfile.js
Comment (akosasante) [inline src/components/UserProfile.js]: If we're using reactstrap as well, you should make sure to install `reactstrap` so that it's in `package.json` otherwise there will be errors when others try to run this.

File: src/index.js
Comment (akosasante) [inline src/index.js]: Where is this file? It's not committed in this PR so when I try to pull your branch and run it, I'm having errors

File: src/index.js
Comment (akosasante) [inline src/index.js]: These aren't used in this file and can be removed

File: package-lock.json
Comment (akosasante) [inline package-lock.json:1]: Make sure to do `git add package-lock.json` and git commit + git push so that this included in the PR; otherwise, right now it's going to be deleting the existing package-lock.json file in main 😱

File: src/App.js
Comment (akosasante) [inline src/App.js]: Make sure to check with the backend team to get a good idea of what fields we expect to get. This looks mostly okay, but I don't think there's a `bio` field and I don't think you can have more than one `role` so it shouldn't be an array.

File: src/components/UserProfile.js
Comment (akosasante) [inline src/components/UserProfile.js]: You pass in an `updateProfileClick` prop, but it's not actually used in this component?

**Review by akosasante — COMMENTED** [review]
Just a few comments (biggest being fixing `package-lock.json` before merging) then you should be good to merge; I'll draft some follow up tickets to handle making sure this component is following similar patterns as some of the existing ones.

File: src/components/UserProfile.js
Comment (akosasante) [inline src/components/UserProfile.js]: We can delete all this commented out code now since it's not being used

File: src/index.js
Comment (akosasante) [inline src/index.js:3]: Feel free to just delete this line and line 8

File: src/App.js
Comment (akosasante) [inline src/App.js]: This is duplicate of line 15 and can be removed

File: src/App.js
Comment (akosasante) [inline src/App.js]: This extra blank space can be undone

File: src/App.js
Comment (akosasante) [reply src/App.js]: @Rjjayy  please check with the team in Slack or check the api docs to verify what fields are actually available: https://musicstore.onrender.com/api-docs/

File: src/App.js
Comment (akosasante) [inline src/App.js:29]: you removed the opening `<>` so this should be removed too

File: src/App.js
Comment (akosasante) [inline src/App.js]: It's fine to have this here for testing; but most likely you'll want to have this as a route, same as the ones above in lines 51-58

File: src/components/UserProfile.js
Comment (akosasante) [inline src/components/UserProfile.js]: All this commented out code can just be removed

File: src/components/getUserProfile.js
Comment (akosasante) [inline src/components/getUserProfile.js:1]: this file can be deleted in favour of using the `useAuth` hook which already does the call to the `/showMe` endpoint

File: src/App.js
Comment (akosasante) [inline src/App.js]: This should not be here, it should be in the Navbar component

File: src/index.js
Comment (akosasante) [inline src/index.js:10]: Feel free to delete this line

**Review by akosasante — COMMENTED** [review]
All the changes look good to me; there's probably some commented out stuff that you could just delete completely at this point though -- I left some comments on those bits.

You'll need to resolve some conflicts before it will let you merge. This is because your _forked_ repo has a `main` [branch](https://github.com/Rjjayy/dd-prac-team4-front/blob/main/src/App.js) that is still a little behind the `main` [branch](https://github.com/Rjjayy/dd-prac-team4-front/blob/main/src/App.js) of the shared repo . You can see the difference if you click those two links to see what `App.js` looks like in each one. 

For `App.js` you can resolve this in the web editor. 

But for `package-lock.js` it is better to follow these instructions to fix it: https://codethedreamteam4-2023.atlassian.net/wiki/spaces/PT4/pages/4194310/Resolving+package-lock.json+conflicts

**Review by akosasante — APPROVED** [review]
Ready to ship! 🚀 🚢

**Comment by Rjjayy** [comment]
User page has Name, Username, email and password.

**Pull request merged**
